# 2.1 Data Collection Report

> **Project:** Titanic Survival Prediction
> **Date:** 2026-03-30
> **CRISP-DM Phase:** 2. Data Understanding
> **Status:** Draft

This notebook documents how each data source was acquired, profiles what was received, maps datasets to data mining goals, and provides reproducible loading instructions.

**Source documents:**
- [1.1 Business Objectives](../docs/crisp-dm/1-business-understanding/1.1-business-objectives.md)
- [1.2 Situation Assessment](../docs/crisp-dm/1-business-understanding/1.2-situation-assessment.md)
- [1.3 Data Mining Goals](../docs/crisp-dm/1-business-understanding/1.3-data-mining-goals.md)

## Setup & Data Loading

In [1]:
import pandas as pd
import os
from pathlib import Path

# Resolve project root: works whether cwd is project root or notebooks/
PROJECT_ROOT = Path(__file__).resolve().parent.parent if "__file__" in dir() else Path.cwd()
if (PROJECT_ROOT / "notebooks").is_dir():
    pass  # cwd is project root
elif (PROJECT_ROOT.parent / "notebooks").is_dir():
    PROJECT_ROOT = PROJECT_ROOT.parent  # cwd is a subdirectory (e.g. notebooks/)

DATA_DIR = PROJECT_ROOT / "data" / "raw" / "titanic"

train = pd.read_csv(DATA_DIR / "train.csv")
test = pd.read_csv(DATA_DIR / "test.csv")
baseline = pd.read_csv(DATA_DIR / "gender_submission.csv")

print("Datasets loaded successfully.")
print(f"  train.csv:             {train.shape[0]} rows × {train.shape[1]} cols")
print(f"  test.csv:              {test.shape[0]} rows × {test.shape[1]} cols")
print(f"  gender_submission.csv: {baseline.shape[0]} rows × {baseline.shape[1]} cols")

Datasets loaded successfully.
  train.csv:             891 rows × 12 cols
  test.csv:              418 rows × 11 cols
  gender_submission.csv: 418 rows × 2 cols


## Data Acquisition Log

All three datasets were downloaded from the [Kaggle Titanic Competition](https://www.kaggle.com/c/titanic) on 2026-03-30. No access issues were encountered — the data is publicly available.

| # | Source | Acquired? | Method | Date | Provider | Issues | Resolution |
|---|--------|-----------|--------|------|----------|--------|------------|
| 1 | train.csv | Yes | Kaggle download | 2026-03-30 | Kaggle | None | — |
| 2 | test.csv | Yes | Kaggle download | 2026-03-30 | Kaggle | None | — |
| 3 | gender_submission.csv | Yes | Kaggle download | 2026-03-30 | Kaggle | None | — |

All planned data sources from the [1.2 Situation Assessment](../docs/crisp-dm/1-business-understanding/1.2-situation-assessment.md) have been acquired. No sources are pending.

## Data Profiling

### Dataset 1: train.csv

- **Source System:** Kaggle Titanic Competition
- **File Format:** CSV (comma-separated, UTF-8)
- **Storage Location:** `data/raw/titanic/train.csv`
- **Date Range:** N/A (single historical event, April 15, 1912)
- **Key Fields:** `PassengerId` (unique identifier), `Survived` (target variable, 0/1)

In [2]:
file_size = os.path.getsize(DATA_DIR / "train.csv")
print(f"File size: {file_size:,} bytes ({file_size / 1024:.0f} KB)")
print(f"Shape: {train.shape[0]} rows × {train.shape[1]} columns")
print(f"Memory usage: {train.memory_usage(deep=True).sum() / 1024:.0f} KB")
print(f"Duplicates: {train.duplicated().sum()}")
print()

# Column summary
summary = pd.DataFrame({
    "Data Type": train.dtypes,
    "Non-Null": train.notnull().sum(),
    "Null Count": train.isnull().sum(),
    "Null %": (train.isnull().sum() / len(train) * 100).round(1),
    "Unique": train.nunique(),
})
summary

File size: 61,194 bytes (60 KB)
Shape: 891 rows × 12 columns
Memory usage: 286 KB
Duplicates: 0



,Data Type,Non-Null,Null Count,Null %,Unique
PassengerId,int64,891,0,0.0,891
Survived,int64,891,0,0.0,2
Pclass,int64,891,0,0.0,3
Name,str,891,0,0.0,891
Sex,str,891,0,0.0,2
Age,float64,714,177,19.9,88
SibSp,int64,891,0,0.0,7
Parch,int64,891,0,0.0,7
Ticket,str,891,0,0.0,681
Fare,float64,891,0,0.0,248


In [3]:
print("Sample values:")
train.head(3)

Sample values:


,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S


In [4]:
train.describe()

,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,891.000000,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,446.000000,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,257.353842,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,1.000000,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,223.500000,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,446.000000,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,668.500000,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,891.000000,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


### Dataset 2: test.csv

- **Source System:** Kaggle Titanic Competition
- **File Format:** CSV (comma-separated, UTF-8)
- **Storage Location:** `data/raw/titanic/test.csv`
- **Date Range:** N/A (single historical event, April 15, 1912)
- **Key Fields:** `PassengerId` (unique identifier, range 892–1309); no `Survived` column

In [5]:
file_size = os.path.getsize(DATA_DIR / "test.csv")
print(f"File size: {file_size:,} bytes ({file_size / 1024:.0f} KB)")
print(f"Shape: {test.shape[0]} rows × {test.shape[1]} columns")
print(f"Memory usage: {test.memory_usage(deep=True).sum() / 1024:.0f} KB")
print(f"Duplicates: {test.duplicated().sum()}")
print()

summary_test = pd.DataFrame({
    "Data Type": test.dtypes,
    "Non-Null": test.notnull().sum(),
    "Null Count": test.isnull().sum(),
    "Null %": (test.isnull().sum() / len(test) * 100).round(1),
    "Unique": test.nunique(),
})
summary_test

File size: 28,629 bytes (28 KB)
Shape: 418 rows × 11 columns
Memory usage: 131 KB
Duplicates: 0



,Data Type,Non-Null,Null Count,Null %,Unique
PassengerId,int64,418,0,0.0,418
Pclass,int64,418,0,0.0,3
Name,str,418,0,0.0,418
Sex,str,418,0,0.0,2
Age,float64,332,86,20.6,79
SibSp,int64,418,0,0.0,7
Parch,int64,418,0,0.0,8
Ticket,str,418,0,0.0,363
Fare,float64,417,1,0.2,169
Cabin,str,91,327,78.2,76


In [6]:
print("Sample values:")
test.head(3)

Sample values:


,PassengerId,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,892,3,"Kelly, Mr. James",male,34.5,0,0,330911,7.8292,NaN,Q
1,893,3,"Wilkes, Mrs. James (Ellen Needs)",female,47.0,1,0,363272,7.0000,NaN,S
2,894,2,"Myles, Mr. Thomas Francis",male,62.0,0,0,240276,9.6875,NaN,Q


In [7]:
test.describe()

,PassengerId,Pclass,Age,SibSp,Parch,Fare
count,418.000000,418.000000,332.000000,418.000000,418.000000,417.000000
mean,1100.500000,2.265550,30.272590,0.447368,0.392344,35.627188
std,120.810458,0.841838,14.181209,0.896760,0.981429,55.907576
min,892.000000,1.000000,0.170000,0.000000,0.000000,0.000000
25%,996.250000,1.000000,21.000000,0.000000,0.000000,7.895800
50%,1100.500000,3.000000,27.000000,0.000000,0.000000,14.454200
75%,1204.750000,3.000000,39.000000,1.000000,0.000000,31.500000
max,1309.000000,3.000000,76.000000,8.000000,9.000000,512.329200


### Dataset 3: gender_submission.csv

- **Source System:** Kaggle Titanic Competition (baseline submission)
- **File Format:** CSV (comma-separated, UTF-8)
- **Storage Location:** `data/raw/titanic/gender_submission.csv`
- **Date Range:** N/A
- **Key Fields:** `PassengerId`, `Survived` (baseline predictions: females=1, males=0)

In [8]:
file_size = os.path.getsize(DATA_DIR / "gender_submission.csv")
print(f"File size: {file_size:,} bytes ({file_size / 1024:.0f} KB)")
print(f"Shape: {baseline.shape[0]} rows × {baseline.shape[1]} columns")
print(f"Duplicates: {baseline.duplicated().sum()}")
print()

summary_baseline = pd.DataFrame({
    "Data Type": baseline.dtypes,
    "Non-Null": baseline.notnull().sum(),
    "Null Count": baseline.isnull().sum(),
    "Null %": (baseline.isnull().sum() / len(baseline) * 100).round(1),
    "Unique": baseline.nunique(),
})
summary_baseline

File size: 3,258 bytes (3 KB)
Shape: 418 rows × 2 columns
Duplicates: 0



,Data Type,Non-Null,Null Count,Null %,Unique
PassengerId,int64,418,0,0.0,418
Survived,int64,418,0,0.0,2


## Cross-Dataset Consistency Checks

In [9]:
# PassengerId disjointness
train_ids = set(train["PassengerId"])
test_ids = set(test["PassengerId"])
baseline_ids = set(baseline["PassengerId"])

print("PassengerId ranges:")
print(f"  train:    {train['PassengerId'].min()} – {train['PassengerId'].max()}")
print(f"  test:     {test['PassengerId'].min()} – {test['PassengerId'].max()}")
print(f"  baseline: {baseline['PassengerId'].min()} – {baseline['PassengerId'].max()}")
print()
print(f"Train/test overlap: {train_ids & test_ids or 'None (disjoint)'}")
print(f"Test/baseline alignment: {'Perfect 1:1 match' if test_ids == baseline_ids else 'MISMATCH'}")
print()

# Column consistency between train and test
train_cols = set(train.columns)
test_cols = set(test.columns)
print(f"Columns in train only: {train_cols - test_cols}")
print(f"Columns in test only:  {test_cols - train_cols}")
print()

# Shared columns — dtype consistency
shared_cols = sorted(train_cols & test_cols)
dtype_mismatches = []
for col in shared_cols:
    if train[col].dtype != test[col].dtype:
        dtype_mismatches.append((col, train[col].dtype, test[col].dtype))
print(f"Dtype mismatches on shared columns: {dtype_mismatches or 'None'}")

PassengerId ranges:
  train:    1 – 891
  test:     892 – 1309
  baseline: 892 – 1309

Train/test overlap: None (disjoint)
Test/baseline alignment: Perfect 1:1 match

Columns in train only: {'Survived'}
Columns in test only:  set()

Dtype mismatches on shared columns: None


## Selection Rationale

| # | Dataset | Data Mining Goal (from 1.3) | Role | Justification |
|---|---------|---------------------------|------|---------------|
| 1 | train.csv | DM1, DM2, DM3, DM4 | Target + Features | Contains the target variable (`Survived`) and all feature columns; used for training and cross-validation |
| 2 | test.csv | DM2 | Features (prediction) | Contains feature columns for 418 passengers; predictions submitted to Kaggle for scoring |
| 3 | gender_submission.csv | DM2 | Baseline reference | Demonstrates submission format and provides the gender-only baseline to beat (~76.5% accuracy) |

### Rejected Alternatives

No alternative data sources were considered. The Kaggle competition provides a fixed dataset; external data augmentation is not permitted by competition rules and is out of scope (see [1.3 Data Mining Goals](../docs/crisp-dm/1-business-understanding/1.3-data-mining-goals.md)).

## Loading & Storage

### Loading Instructions

All datasets use standard CSV format with comma delimiters and UTF-8 encoding. Default `pandas.read_csv()` parameters work without modification.

```python
import pandas as pd

train = pd.read_csv("data/raw/titanic/train.csv")
test = pd.read_csv("data/raw/titanic/test.csv")
baseline = pd.read_csv("data/raw/titanic/gender_submission.csv")
```

**Dependencies:** `pandas`
**Notes:** No special handling required (no custom delimiters, encodings, or parse dates).

### Storage Overview

| Dataset | Location | Format | Size | Version Controlled |
|---------|----------|--------|------|--------------------|
| train.csv | `data/raw/titanic/train.csv` | CSV | 60 KB | Git (data < 1 MB) |
| test.csv | `data/raw/titanic/test.csv` | CSV | 28 KB | Git (data < 1 MB) |
| gender_submission.csv | `data/raw/titanic/gender_submission.csv` | CSV | 3 KB | Git (data < 1 MB) |

All files are small enough to version-control directly in Git. DVC is not needed for this project.

## Key Observations from Initial Profiling

1. **Missingness pattern consistent across train/test:** Age ~20% missing in both sets; Cabin ~77-78% missing in both sets. Imputation strategy must handle both.
2. **Test set has 1 missing Fare:** train.csv has no missing Fare values, but test.csv has 1 (PassengerId 1044). Must handle in preprocessing.
3. **PassengerId ranges are disjoint:** train (1–891), test (892–1309). No overlap, confirming clean train/test split.
4. **Ticket column is messy:** 681 unique values across 891 rows — many tickets are shared by family groups. Potential feature engineering signal (shared ticket = traveling together).
5. **Name column contains titles:** "Mr.", "Mrs.", "Miss.", "Master." embedded in names. Key feature engineering opportunity.